slash-commands/moderation/timeout.js

```diff
@@ -0,0 +1,181 @@
+      option
+        .setName('user')
+        .setDescription('The user to timeout')
+        .setRequired(true)
+    )
+    .addStringOption((option) =>
+      option
+        .setName('duration')
+        .setDescription(
+          'How long the user should be timed out as <number><s|m|h|d|w>'
+        )
+        .setRequired(true)
+    )
+    .addStringOption((option) =>
+      option
+        .setName('reason')
+        .setDescription('The reason for the timeout')
+        .setRequired(true)
+    ),
+
+  async execute(interaction) {
+    let toTimeout;
+    interaction.guild.members
+      .fetch(await interaction.options.getUser('user'))
+      .then((result) => (toTimeout = result));
+    const duration = await interaction.options.getString('duration');
+    const reason = await interaction.options.getString('reason');
+
+    if (canTimeout(interaction, toTimeout, duration, reason)) {
+      try {
+        await timeoutUser(interaction, toTimeout, duration, reason);
+      } catch (e) {
+        console.log(e);
+        interaction.reply(
+          'Something went wrong while trying to timeout the user.'
+        );
+      }
+      await auditLogTimeout(interaction, toTimeout, duration, reason);
+      await recordTimeoutInDB(interaction, toTimeout, duration, reason);
+      try {
+        await dmTheUser(interaction, toTimeout, duration, reason);
+      } catch (e) {
+        console.log(e);
+        interaction.reply(`${e.name}: ${e.message}`);
+      }
+    }
+  },
+};
+
+// Validates data.
+function canTimeout(interaction, toTimeout, duration, reason) {
+  if (toTimeout === interaction.member) {
+    interaction.reply('You cannot timeout yourself.');
+    return false;
+  }
+  if (
+    toTimeout.roles.cache.some(
+      (role) =>
+        role.name === 'Moderator' ||
+        role.name === 'Admin' ||
+        role.name === 'Super Admin'
+    )
+  ) {
+    interaction.reply('You cannot timeout a moderator or admin.');
+    return false;
+  }
+  if (
+    toTimeout.communicationDisabledUntilTimestamp !== null &&
+    toTimeout.communicationDisabledUntil > Date.now()
+  ) {
+    interaction.reply('This user is already timed out.');
+    return false;
+  }
+
+  // Check if duration is valid.
+  if (!duration.match(/^\d+.?\d*[smhdw]$/)) {
+    interaction.reply(
+      'Please provide a valid duration. Format should be `<number><s|m|h|d|w>`.'
+    );
+    return false;
+  }
+  if (ms(duration) > ms('28d')) {
+    interaction.reply('Please enter a duration less than 28 days.');
+    return false;
+  }
+
+  // Check if reason is of a valid length.
+  if (!verifyReasonLength(reason, interaction)) return false;
+
+  return true;
+}
+
+// Puts a user in timeout.
+async function timeoutUser(interaction, toTimeout, duration, reason) {
+  const durationInMs = ms(duration);
+  toTimeout.timeout(durationInMs, reason);
+
+  await interaction.reply(`${toTimeout} has been timed out for ${duration}.`);
+}
+
+// Sends message to #audit-logs.
```

Comment:
I think it's better if we can use the function sendToAuditLogsChannel from helpers to send this message to the audit logs instead. The audit log recording is fairly similar between the commands.